src/dcat-us/compile-dcat-feed.ts

```diff
@@ -9,7 +9,11 @@ export function compileDcatFeedEntry(dataset: any, feedTemplate: DcatDatasetTemp
     const defaultDataset = {
       '@type': 'dcat:Dataset'
     };
-    const dcatDataset = Object.assign({}, defaultDataset, adlib(feedTemplate, dataset, feedTemplateTransforms));
+    const dcatDataset = Object.assign(
+      {}, 
+      defaultDataset, 
+      adlib(feedTemplate, { ...dataset?.properties, ...dataset?.geometry }, feedTemplateTransforms)
+    );
```

Comment:
Too much going on here to easily understand.  Can you move to a helper and make it more clear?

Reply:
Agreed! Moved it into `generateDcatDataset()` function and updated variable names to make it more clear.